caravel/models.py

```diff
@@ -436,9 +436,10 @@ def grains(self):
             if self.sqlalchemy_uri.startswith(db_type):
                 return grains
 
-    def dttm_converter(self, dttm):
+    def dttm_converter(self, dttm, tf=None):
         """Returns a string that the database flavor understands as a date"""
-        default = "'{}'".format(dttm.strftime('%Y-%m-%d %H:%M:%S.%f'))
+        tf = tf or '%Y-%m-%d %H:%M:%S.%f'
+        default = "'{}'".format(dttm.strftime(tf))
         iso = dttm.isoformat()
         d = {
```

Comment:
i think this previous approach had shortcomings to start with. It assumed that the casting required could be defined on a per-database basis where it cannot since even within a single table you could represent dates in different fashion (epoch, string, native DATETIME, DATE or TIMESTAMP types...).